src/main/java/org/swisspush/reststorage/RedisStorage.java

```diff
@@ -252,29 +267,30 @@ public void recomposeLuaScript() {
 
         /**
          * Load the get script into redis and store the sha in the class member sha.
-         * @param redisCommand the redis command that should be executed, after the script is loaded.
+         *
+         * @param redisCommand     the redis command that should be executed, after the script is loaded.
          * @param executionCounter a counter to control recursion depth
          */
         public void loadLuaScript(final RedisCommand redisCommand, int executionCounter) {
             final int executionCounterIncr = ++executionCounter;
 
             // check first if the lua script already exists in the store
-            redisClient.scriptExists(this.sha, resultArray -> {
-                if(resultArray.failed()){
+            redisAPI.script(Arrays.asList("exists", this.sha), resultArray -> {
+                if (resultArray.failed()) {
                     log.error("Error checking whether lua script exists", resultArray.cause());
                     return;
                 }
-                Long exists = resultArray.result().getLong(0);
+                Long exists = resultArray.result().get(0).toLong();
                 // if script already
-                if(Long.valueOf(1).equals(exists)) {
+                if (Long.valueOf(1).equals(exists)) {
                     log.debug("RedisStorage script already exists in redis cache: " + luaScriptType);
                     redisCommand.exec(executionCounterIncr);
                 } else {
                     log.info("load lua script for script type: {} logoutput: {}", luaScriptType, logoutput);
-                    redisClient.scriptLoad(script, stringAsyncResult -> {
-                        String newSha = stringAsyncResult.result();
+                    redisAPI.script(Arrays.asList("load", script), stringAsyncResult -> {
+                        String newSha = stringAsyncResult.result().toString();

```

Comment:
If [stringAsyncResult.failed()](https://vertx.io/docs/apidocs/io/vertx/core/AsyncResult.html#failed--) reports `true` we will call `.toString()` on a null-value. Result: NullPointerException

Reply:
Good catch, wasn't handled in the "old" version too I believe.

Reply:
Hehe. That looks funny :) Would it work better if pasted into a code block maybe? Like:
```
` ` `patch
diff --git a/src/Blah.java
--- a/src/Blah.java
+++ b/src/Blah.java
@@ -283,11 +283,17 @@
 foo();
-old();
+new();
 bar();
` ` `
```

This would render as:
```patch
diff --git a/src/main/java/org/swisspush/reststorage/RedisStorage.java b/src/main/java/org/swisspush/reststorage/RedisStorage.java
--- a/src/main/java/org/swisspush/reststorage/RedisStorage.java (revision 1c5bde0)
+++ b/src/main/java/org/swisspush/reststorage/RedisStorage.java (date 1642730777059)
@@ -283,11 +283,17 @@
 foo();
-old();
+new();
 bar();
```